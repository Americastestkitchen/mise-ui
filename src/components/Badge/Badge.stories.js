import React from 'react';
import { Badge } from '../Badge';

export default {
  title: 'Components|Badge',
  component: Badge,
}

export const ATKBadge = () => <Badge type='atk'/>

export const CIOBadge = () => <Badge type='cio'/>

export const CCOBadge = () => <Badge type='cco'/>

export const KidsBadge = () => <Badge type='kids'/>
