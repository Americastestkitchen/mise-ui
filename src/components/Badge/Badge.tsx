import React from 'react';
import styled from 'styled-components';
import { color } from '../../styles';
import {
  ATKBrandIcon,
  CIOBrandIcon,
  CCOBrandIcon,
  KidsBrandIcon,
  SchoolBrandIcon,
  ShopBrandIcon,
} from './svgs';

const StyledBadge = styled.svg`
  width: 2.5rem;
  height: 2.5rem;
`;

export type BadgeType = 'atk' | 'cio'| 'cco'| 'kids'| 'school'| 'shop';

export type BadgeProps = {
  className?: string;
  fill?: string;
  type: BadgeType;
}

const determineType = (type: BadgeType, fill = `${color.transparentBlack}`) => {
  const types = {
    atk: ATKBrandIcon,
    cio: CIOBrandIcon,
    cco: CCOBrandIcon,
    kids: KidsBrandIcon,
    school: SchoolBrandIcon,
    shop: ShopBrandIcon,
  };
  const El = types[type];
  return El && <El fill={fill} />;
};

const determineBadgeText = (type: BadgeType) => {
  const types = {
    atk: 'America’s Test Kitchen',
    cio: 'Cook’s Illustrated',
    cco: 'Cook’s Country',
    kids: 'ATK Kids',
    school: 'Cooking School',
    shop: 'ATK Shop',
  };
  const El = types[type];
  return El;
};

/**
 * Badges exist for brand logos (cc, ci, atk, kids) and also other destination
 * identities like shop.
 * <br>
 * Logo badges appear on cards in mixed brand scenarios, like search and
 * do not appear when only one brand’s content is showing
 *  Badges on a card do not go anywhere, but the badges that
 *  appear in the browse bar *are* interactive and function as facets
 */

export default function Badge({
  className,
  fill,
  type,
}: BadgeProps) {
  return (
    <StyledBadge
      data-testid={`${type}-badge`}
      role="img"
      aria-label={determineBadgeText(type)}
      className={`badge${className ? ` ${className}` : ''}`}
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 25 25"
    >
      {determineType(type, fill)}
    </StyledBadge>
  );
}
