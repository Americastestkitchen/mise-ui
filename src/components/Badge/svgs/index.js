import ATKBrandIcon from './ATKBrandIcon';
import CIOBrandIcon from './CIOBrandIcon'
import CCOBrandIcon from './CCOBrandIcon';
import KidsBrandIcon from './KidsBrandIcon';

export {
  ATKBrandIcon,
  CIOBrandIcon,
  CCOBrandIcon,
  KidsBrandIcon,
};
