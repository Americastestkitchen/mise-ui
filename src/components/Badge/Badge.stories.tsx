import React from 'react';
import type { ComponentMeta, ComponentStory } from '@storybook/react';

import Badge, { BadgeProps } from './Badge';

const type: any = {
  options: ['atk', 'cio', 'cco', 'kids', 'school', 'shop'],
  control: { type: 'inline-radio' },
};

export default {
  title: 'Components/Badge',
  component: Badge,
  argTypes: { type },
} as ComponentMeta<typeof Badge>;

const defaultArgs: BadgeProps = {
  type: 'atk',
}
const Template: ComponentStory<typeof Badge> = props => <Badge {...props} />;

export const badges = Template.bind({});
badges.args = { ...defaultArgs };