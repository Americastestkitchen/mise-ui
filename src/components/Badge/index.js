import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { color } from '../../styles';
import { ATKBrandIcon, CIOBrandIcon, CCOBrandIcon, KidsBrandIcon } from './svgs';

const StyledBadge = styled.div`
  width: 2.5rem;
  height: 2.5rem;
`

const determineType = (type, fill) => ({
  "atk": <ATKBrandIcon fill={fill} />,
  "cio": <CIOBrandIcon fill={fill} />,
  "cco": <CCOBrandIcon fill={fill} />,
  "kids": <KidsBrandIcon fill={fill} />,
})[type]

/**
 * Badges exist for brand logos (cc, ci, atk, kids) and also other destination identities like shop.
 * <br>
 * Logo badges appear on cards in mixed brand scenarios, like search and do not appear when only one brandâ€™s content is showing
 *  Badges on a card do not go anywhere, but the badges that appear in the browse bar *are* interactive and function as facets
 */
export function Badge({
  className,
  fill,
  type,
}) {
  return (
    <StyledBadge className={className}>
      {determineType(type, fill)}
    </StyledBadge>
  );
}

Badge.propTypes = {
  className: PropTypes.string,
  fill: PropTypes.string,
  type: PropTypes.oneOf(['atk', 'cio', 'cco', 'kids']).isRequired,
}

Badge.defaultProps = {
  className: '',
  fill: `${color.transparentBlack}`,
}
