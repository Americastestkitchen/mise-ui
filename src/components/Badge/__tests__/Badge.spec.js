import React from 'react';
import Badge from '../Badge';
import renderer from 'react-test-renderer';

it('renders an atk badge', () => {
  const atkBadge = renderer.create(<Badge type='atk'/>).toJSON();
  expect(atkBadge).toMatchSnapshot();
});


it('renders an cio badge', () => {
  const cioBadge = renderer.create(<Badge type='cio'/>).toJSON();
  expect(cioBadge).toMatchSnapshot();
});


it('renders an cco badge', () => {
  const ccoBadge = renderer.create(<Badge type='cco'/>).toJSON();
  expect(ccoBadge).toMatchSnapshot();
});