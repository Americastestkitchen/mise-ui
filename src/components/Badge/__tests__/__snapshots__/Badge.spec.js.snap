// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components renders a cco badge 1`] = `
<svg
  aria-label="cco brand icon"
  className="sc-AxjAm fBdxjG"
  role="img"
  viewBox="0 0 25 25"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <clipPath
      id="cco-brand-icon-a"
    >
      <path
        d="M0 0h14.1v16.226H0z"
        fill="none"
      />
    </clipPath>
    <clipPath
      id="cco-brand-icon-b"
    >
      <path
        d="M0 0h3.614v3.635H0z"
        fill="none"
      />
    </clipPath>
  </defs>
  <circle
    cx="12.5"
    cy="12.5"
    fill="rgba(0, 0, 0, 0.7)"
    r="12.5"
  />
  <g>
    <g>
      <path
        d="M5 4h14.1v16.226H5z"
        fill="none"
      />
      <g
        clipPath="url(#cco-brand-icon-a)"
        transform="translate(5 4)"
      >
        <path
          d="M12.309 16.226a2.058 2.058 0 01-.832-.526 2.614 2.614 0 01-.462-.689.492.492 0 00-.044-.081.027.027 0 00-.016-.006.02.02 0 00-.014.007 1.9 1.9 0 00-.015.7l.023.227a.86.86 0 01.007.363.019.019 0 01-.015.007c-.091 0-.3-.35-.363-.921a6.212 6.212 0 01-.016-.178c-.016-.21-.034-.447-.11-.506a.024.024 0 00-.015 0 1.2 1.2 0 00-.228.063l-.11.035a1.764 1.764 0 01-.459.1.178.178 0 01-.162-.061.006.006 0 010-.005.874.874 0 01.237-.135c.213-.1.478-.233.487-.324a.049.049 0 00-.023-.045 5.886 5.886 0 00-1.013-.394c-.192-.065-.372-.125-.52-.18A7.976 7.976 0 015.905 11.9a6.576 6.576 0 01-.9-1.131c-.347-.513-.647-.957-1-.957a.44.44 0 00-.114.015 1.3 1.3 0 00-.594.493 2.611 2.611 0 01.03-.969 2.277 2.277 0 01.59-1.059.911.911 0 00-.265-.036c-.286 0-.813.122-1.136.936a2.513 2.513 0 00-.069 1.429h-.005a2.619 2.619 0 01-.622-.836 2.194 2.194 0 01.2-2.169 2.359 2.359 0 012.068-1.148 1.908 1.908 0 01.294.02 1.511 1.511 0 00-1.106-.476 2.242 2.242 0 00-1.166.375A2.449 2.449 0 00.966 8.179 3.38 3.38 0 00.978 9.26a2.668 2.668 0 01-.947-2.479 2.936 2.936 0 011.831-2.274 2.888 2.888 0 011.1-.223 3.5 3.5 0 012.257.945 2.232 2.232 0 00-.353-.892 2.093 2.093 0 00-1.85-.892 3.027 3.027 0 00-2.658 1.4 3.2 3.2 0 013.45-3.064 4.625 4.625 0 01.481.025A2.869 2.869 0 016.539 3.2a7.926 7.926 0 01.855 2.286c.2.791.362 1.417.748 1.417a1.41 1.41 0 001.133-.523 2.47 2.47 0 00.461-1.872 1 1 0 00.551-.552 1.331 1.331 0 00-.069-1.181c0 1.261-.583 1.4-.834 1.4A.73.73 0 019.07 4.1a.624.624 0 01-.305-.383.953.953 0 01.067-.731 1.53 1.53 0 01.517-.534S9.17 2.2 8.371 2a2.16 2.16 0 01.95-.23h.045s-.179-.319-1.033-.656a1.254 1.254 0 01.518-.083c.07 0 .145 0 .221.009a1.31 1.31 0 01.695.221.355.355 0 01-.009-.039 4.418 4.418 0 01-.04-.2c0-.022-.007-.045-.011-.068-.025-.168-.055-.359.064-.416A.279.279 0 019.89.507c.116 0 .173.107.206.187.013.03.023.06.032.087.02.058.035.1.055.1.031-.01.039-.1.048-.21.016-.188.038-.444.193-.486a.353.353 0 01.091-.013c.187 0 .24.194.276.323.019.068.033.121.06.122.074 0 .138-.107.212-.231l.047-.078c.1-.16.216-.308.387-.308a.345.345 0 01.122.024c.368.14.257.437.177.653-.039.105-.07.189-.034.223a.116.116 0 00.082.029.76.76 0 00.256-.077l.037-.015a.963.963 0 01.353-.092.372.372 0 01.293.136c.232.265-.016.557-.255.839a4.558 4.558 0 00-.17.208 2.4 2.4 0 00-1.43-.568 1.052 1.052 0 00-.37.057A2.939 2.939 0 0112.2 2.2a5.2 5.2 0 011.39 1.751 6.553 6.553 0 01.453 3.549c-.377 3.367-3.237 4.72-4.314 5.229a2.514 2.514 0 00-.411.216.123.123 0 00-.048.11c.025.2.552.479 1.062.745a7.087 7.087 0 01.715.4 1.213 1.213 0 01.378.647 2.528 2.528 0 00.226.523 4.387 4.387 0 00.528.663c.089.1.148.16.14.178s-.004.015-.01.015zM7.013 9.42a2.626 2.626 0 00.144 2.526 7.122 7.122 0 013.169-4.726 4.7 4.7 0 00-3.315 3.792 4.027 4.027 0 01.4-2.191 2.005 2.005 0 00-1.637 1.53c-.007-.8.423-1.242 1.219-1.775a1.469 1.469 0 00-.393-.05 1.819 1.819 0 00-1.355.566A2.5 2.5 0 017.13 7.446h-.006a2.048 2.048 0 00-1.549.628A2.348 2.348 0 005 9.711a2.6 2.6 0 011.271-.789 1.8 1.8 0 00-.442 2.2 2.67 2.67 0 011.185-1.7zm4.742-.8c-.671 1.736-2.382 1.838-3.4 1.9l-.155.009a3.382 3.382 0 001.254.248 2.414 2.414 0 001.765-.682 1.97 1.97 0 00.539-1.475zM11.7 6.649C11.383 8.1 10.679 8.644 9.12 9.39a2.969 2.969 0 002.294-1.081 1.815 1.815 0 00.286-1.66zm.416-1.794a4.3 4.3 0 01.646 2.007 3.1 3.1 0 00.173-2.129 1.572 1.572 0 00-.756-.984 3.336 3.336 0 01.53 2.151 2.306 2.306 0 00-.857-1.444 2.271 2.271 0 01-.3 1.288 2.757 2.757 0 00-.369-1.489 3.045 3.045 0 01-.746 1.5 1.889 1.889 0 00-.171-1.265 2.767 2.767 0 01-.1 1.738v.014c.665-.1 1.054-1.087 1.139-1.548a5.257 5.257 0 01.118 1.535 1.562 1.562 0 00.7-1.375zm-9.139.491a2.6 2.6 0 012 .914 1.764 1.764 0 00-1.758-1.241 2.394 2.394 0 00-1.643.647 3.411 3.411 0 011.405-.32zm1.039-2.52c-.059 0-.119 0-.178.005a3.051 3.051 0 012.437 2.7 2.317 2.317 0 00-.423-1.94 2.372 2.372 0 00-1.831-.765zm6.481-.82a.2.2 0 10.2.2.207.207 0 00-.013-.069.087.087 0 01-.073.043.086.086 0 01-.086-.085.083.083 0 01.043-.072.2.2 0 00-.068-.016z"
          fill="#fefefe"
        />
      </g>
    </g>
    <g>
      <path
        d="M11.866 18.207h3.614v3.635h-3.614z"
        fill="none"
      />
      <g
        clipPath="url(#cco-brand-icon-b)"
        transform="translate(11.866 18.207)"
      >
        <path
          d="M3.155 2.91c-.385-.118-.865-.1-1.071-.5a5.2 5.2 0 010-1.976A8.592 8.592 0 011.123 0a6.676 6.676 0 01.257 2.583c-.036.131-.453.138-.8.285s-.621.4-.577.423a7.4 7.4 0 00.844-.144 2.3 2.3 0 01.6-.054c.023.016-.01.1-.021.253-.009.118-.037.288 0 .288s.146-.143.294-.343c.031-.043.1-.177.147-.195s.48.056.8.092c.425.048.9.183.932.149s-.062-.308-.454-.428"
          fill="#fefefe"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`components renders a cio badge 1`] = `
<svg
  aria-label="cio brand icon"
  className="sc-AxjAm fBdxjG"
  role="img"
  viewBox="0 0 25 25"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12.5"
    cy="12.5"
    fill="rgba(0, 0, 0, 0.7)"
    r="12.5"
  />
  <path
    d="M13.323 5C8.522 5 5.31 7.918 5.022 12.389c-.3 4.546 2.644 7.937 7.436 8.28 3.089.243 5.346-.84 6.652-3.134l-.352-.23a5.636 5.636 0 01-5.458 2.846c-3.688-.284-5.843-3.3-5.543-7.878.291-4.465 2.343-6.759 5.566-6.756 2.241 0 4.366 2.11 4.354 3.8a.7.7 0 01.027.259h.351v-.69c.011-1.143.053-2.346.108-3.68h-.324c-.108.354-.213.532-.433.546A19.024 19.024 0 0013.323 5z"
    fill="#fff"
  />
</svg>
`;

exports[`components renders a kids badge 1`] = `
<svg
  aria-label="kids brand icon"
  className="sc-AxjAm fBdxjG"
  role="img"
  viewBox="0 0 25 25"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <clipPath
      id="kids-brand-icon-clip-path"
    >
      <rect
        fill="none"
        height="8.347"
        width="18.46"
      />
    </clipPath>
  </defs>
  <circle
    cx="12.5"
    cy="12.5"
    fill="rgba(0, 0, 0, 0.7)"
    id="Ellipse_1"
    r="12.5"
  />
  <g
    transform="translate(12.23 12.173)"
  >
    <g
      clipPath="url(#kids-brand-icon-clip-path)"
      fill="#fff"
      transform="translate(-9.23 -4.173)"
    >
      <path
        d="M28.313 7.341c.03.03.335.122 1.158-.122s1.127-2.041.762-2.772a1.481 1.481 0 00-1.95-.548c-.091.03-.091.244-.061.7s.03 1.8.061 2.467c0 .061 0 .213.03.213m1.462-4.935c2.1.366 2.711 2.985 1.645 4.9a2.822 2.822 0 01-2.22 1.62 12.063 12.063 0 01-1.736.152c-.487 0-.548-.061-.579-.548-.091-1.675 0-3.381-.061-5.057a1.25 1.25 0 00-.03-.335.6.6 0 01.457-.609l.853-.122a5.026 5.026 0 011.645 0"
        transform="translate(-18.626 -1.584)"
      />
      <path
        d="M48.869.7a6.306 6.306 0 01-.426.914.426.426 0 01-.335.152 3.138 3.138 0 01-.487-.183 4.691 4.691 0 00-.792-.213.609.609 0 00-.609.335.579.579 0 00.244.731l.548.213 1.066.3a1.864 1.864 0 01.457 3.138 3.6 3.6 0 01-4.082.366c-.091-.061-.335-.213-.274-.4s.4-.762.457-.975.122-.426.4-.244a2.821 2.821 0 001.554.609 1.047 1.047 0 00.944-.426.426.426 0 00-.122-.579 1.4 1.4 0 00-.548-.213l-1.193-.265a1.8 1.8 0 01-1.1-2.467 1.645 1.645 0 01.4-.579c1.066-.975 2.8-1.31 3.838-.3.03.03.061.061.061.091"
        transform="translate(-30.713)"
      />
      <path
        d="M5.331 9.908c.122.213-.152.3-.609.548s-.731.426-.883.274L3.26 10a7.991 7.991 0 01-.731-.914 6.074 6.074 0 00-.67-.975l-.274.366a.883.883 0 00-.061.518c0 .213 0 1.036.03 1.554s-.03.457-.4.487H.7c-.579.091-.609.091-.64-.457C0 8.812.03 6.984 0 5.187c0-.548.03-.64.548-.7l.762-.061c.152.03.152.122.152.4s-.03 1.34-.03 1.858c.426-.4.609-.67 1.005-1.1L3.5 4.273c.366-.426.64-.274 1.036-.091s.335.152.426.274a29.994 29.994 0 01-1.95 2.559c.857 1.218 2.228 2.711 2.319 2.893"
        transform="translate(0 -2.78)"
      />
      <path
        d="M20.384 14.6c0 .366-.091.457-.457.518l-.64.091c-.335.03-.457-.061-.457-.4q-.03-1.8-.03-3.595c0-.548.091-.366.762-.426s.822-.4.762.792v.853c0 .3.03 1.462.03 2.193"
        transform="translate(-13.073 -7.444)"
      />
      <path
        d="M19.342 4.724c-.487.061-.853-.3-.853-.883a.883.883 0 01.883-.975.792.792 0 01.944.792 1.188 1.188 0 01-.975 1.066"
        transform="translate(-12.854 -1.983)"
      />
    </g>
  </g>
</svg>
`;

exports[`components renders an atk badge 1`] = `
<svg
  aria-label="atk brand icon"
  className="sc-AxjAm fBdxjG"
  role="img"
  viewBox="0 0 25 25"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12.5"
    cy="12.5"
    fill="rgba(0, 0, 0, 0.7)"
    r="12.5"
  />
  <path
    d="M6.035 16l.41-1.6h2.06l.4 1.6h1.59l-1.96-7.22h-1.99l-2 7.22zm2.15-2.71h-1.44l.69-3.12h.02zm1.85-4.51v1.2h1.68V16h1.51V9.98h1.68v-1.2zm7.01 3.09h-.02V8.78h-1.51V16h1.51v-3.7h.02l1.73 3.7h1.7l-2.02-3.98 2.02-3.24h-1.66z"
    fill="#fff"
  />
</svg>
`;
